packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -0,0 +1,729 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:ui' show Path;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'theme.dart';
+import 'theme_data.dart';
+
+/// Applies a slider theme to descendant [Slider] widgets.
+///
+/// A slider theme describes the colors and shape choices of the slider
+/// components.
+///
+/// Descendant widgets obtain the current theme's [SliderThemeData] object using
+/// [SliderTheme.of]. When a widget uses [SliderTheme.of], it is automatically
+/// rebuilt if the theme later changes.
+///
+/// See also:
+///
+///  * [SliderThemeData], which describes the actual configuration of a slider
+///    theme.
+///  * [SliderComponentShape], which can be used to create custom shapes for
+///    the slider thumb and value indicator.
+class SliderTheme extends InheritedWidget {
+  /// Applies the given theme [data] to [child].
+  ///
+  /// The [data] and [child] arguments must not be null.
+  const SliderTheme({
+    Key key,
+    @required this.data,
+    @required Widget child,
+  }) : assert(child != null),
+       assert(data != null),
+       super(key: key, child: child);
+
+  /// Specifies the color and shape values for descendant slider widgets.
+  final SliderThemeData data;
+
+  /// Returns the data from the closest [SliderTheme] instance that encloses
+  /// the given context.
+  ///
+  /// Defaults to the ambient [ThemeData.sliderTheme] if there is no
+  /// [SliderTheme] in the given build context.
+  ///
+  /// Typical usage is as follows:
```

Comment:
sample code should be in a `## Sample code` section.

Reply:
Done (in another PR)